algorithms/insertion_ll/linked_list.cpp

```diff
@@ -0,0 +1,11 @@
+void insert(lnode*& curr, int val) {
+struct lnode* val = (struct lnode*) malloc(sizeof(struct lnode)); 
```

Comment:
The allocation of a struct `lnode` isn't required. We're using `new lnode` here.